### PR TITLE
Rev1 - ImageUpload - Adding Image Confirmation

### PR DESCRIPTION
**Pull request by jeff-rey-wang**
### What problem does this solve?

Context: User needs to confirm ML image classification output to prevent faulty entries from being logs

### How does this solve it?
Added confirmation menu for users to confirm if the output is correct

### CODE REVIEW

File: src/utrition/src/components/imageupload/imageupload.js [hunk 1]
```diff
@@ -5,6 +5,9 @@ import "./imageupload.css";
 const ImageUpload = () => {
   const [image, setImage] = useState("");
   const [responseData, setResponseData] = useState("");
+  const [showConfirmationDialog, setShowConfirmationDialog] = useState(false);
+  const [processing, setProcessing] = useState(false)
```
Comment (jeff-rey-wang) [inline src/utrition/src/components/imageupload/imageupload.js:9]: New state variables

File: src/utrition/src/components/imageupload/imageupload.js [hunk 2]
```diff
@@ -14,11 +17,81 @@ const ImageUpload = () => {
     upload_type: "image",
   };
 
+  // New header
```
Comment (jeff-rey-wang) [inline src/utrition/src/components/imageupload/imageupload.js:20]: Added unique header for backend to handle (let me know if this needs to be changed)

File: src/utrition/src/components/imageupload/imageupload.js [hunk 3]
```diff
@@ -14,11 +17,81 @@ const ImageUpload = () => {
...
+  const headers_confirm = {
+    upload_type: "confirm",
+  };
+
+  
   const handleClick = (event) => {
     fileInput.current.click();
   };
 
+  // Doesn't send any response, set food to 'correct', close confirmation
```
Comment (jeff-rey-wang) [inline src/utrition/src/components/imageupload/imageupload.js:30]: No responses sent if food is correct, just update the corresponding state variables, and close menus

File: src/utrition/src/components/imageupload/imageupload.js [hunk 4]
```diff
@@ -14,11 +17,81 @@ const ImageUpload = () => {
...
+  function handleCorrectData() {
+    setCorrectFood(true);
+    setShowConfirmationDialog(false);
+  }
+
+  // Sends response to /upload, with new header;
+  // set food to 'incorrect', close confirmation
+  // populate and show error message
```
Comment (jeff-rey-wang) [inline src/utrition/src/components/imageupload/imageupload.js:38]: Sends response to backend for a faulty classification, sets the corresponding state variables to reflect a bad classification

File: src/utrition/src/components/imageupload/imageupload.js [hunk 5]
```diff
@@ -14,11 +17,81 @@ const ImageUpload = () => {
...
+      .catch((error) => console.log(error));
+      setResponseData({
+        error_msg: "Please try again or use a different image."
+      })
+      setShowConfirmationDialog(false);
+  }
+
+  // Shows confirmation if in the correct state;
+  // user will either deem the output correct/incorrect;
+  // handled 2 different ways
```
Comment (jeff-rey-wang) [inline src/utrition/src/components/imageupload/imageupload.js:59]: Conditionally renders confirmation menu to prompt user to confirm their classification

File: src/utrition/src/components/imageupload/imageupload.js [hunk 6]
```diff
@@ -14,11 +17,81 @@ const ImageUpload = () => {
...
+                No
+              </button>
+            </div>
+          </div>
+        </div>
+    );
+  };
+
   async function getData() {
+    // clear response data, enter 'processing' state and predict 'incorrect' food
```
Comment (jeff-rey-wang) [inline src/utrition/src/components/imageupload/imageupload.js:90]: Update relevant state variables before and after a response is received

**Review by jsrebrnjak — APPROVED** [review]

**Pull request merged**